File: r/sparklabs/batch/sales/sql/distinct_customer.R
source("r/sparklabs/batch/sales/sql/utils.R")

customer_count <- collect(run_sql("
  SELECT COUNT(customer_id) AS total_customers
  FROM orders
"))$total_customers[[1]]

cat(sprintf("Total distinct customers: %s\n", customer_count))
cat("Most Expensive Product:\n")
showDF(run_sql("
  SELECT *
  FROM products
  ORDER BY price DESC
  LIMIT 1
"), truncate = FALSE)

File: python/sparklabs/batch/sales/sql/distinct_customer.py
from sparklabs.batch.sales.sql import utils


def run() -> None:
    customer_count = utils.sql(
        """
        SELECT COUNT(customer_id) AS total_customers
        FROM orders
        """
    ).collect()[0][0]
    expensive_product = utils.sql(
        """
        SELECT *
        FROM products
        ORDER BY price DESC
        LIMIT 1
        """
    )

    print(f"Total distinct customers: {customer_count}")
    print("Most Expensive Product:")
    expensive_product.show(truncate=False)


if __name__ == "__main__":
    run()
